### PR TITLE
[Payments] Responsive header

### PR DESCRIPTION
**Pull request by peschina**
### Ticket:

- [16652](https://dev.azure.com/OGCIO-Digital-Services/Digital%20Services%20Programme/_workitems/edit/16652)

### Description

Moved `Header` component to components folder and updated it to make it responsive. An hamburger button is available on mobile and opens a side menu. This is currently available with js enabled only.
Header used in info route will be changed to use the common header in a separate pr. Responsiveness of the recently added building blocks tiles will be addressed separately as well

## Type

- [ ] **Dependency upgrade**
- [ ] **Bug fix**
- [x] **New feature**
- [ ] **Dev change**

### Checklist:

- [ ] I have added tests that prove my fix is effective or that my feature works

### Screenshots:

Public Servant:
localhost_3001_en_paymentSetup(iPhone 12 Pro)
localhost_3001_en_paymentSetup(iPhone 12 Pro) (1)

Citizen:
localhost_3001_en_citizen_transactions(iPhone 12 Pro)

### CODE REVIEW

File: apps/payments/app/components/HamburgerMenu/HamburgerMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+          <Link
+            className={`govie-link govie-link--no-underline`}
+            style={{
+              fontWeight: locale === "en" ? "bold" : "normal",
+              color: ds.colours.ogcio.darkGreen,
+            }}
+            href={"/en/" + pathSlice}
+            prefetch={false}
+          >
+            English
```
Comment (peschina) [inline apps/payments/app/components/HamburgerMenu/HamburgerMenu.tsx:107]: Should the language names be translated as well?

Comment (alfonsograziano) [reply apps/payments/app/components/HamburgerMenu/HamburgerMenu.tsx:109]: We can keep it like this for now, not sure if we will ever need to translate it :)

File: apps/payments/app/components/Header/LanguageSwitch.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import ds from "design-system";
+import { headers } from "next/headers";
+
+export default () => {
+  const pathSlice = headers().get("x-pathname")?.split("/") ?? [];
+  const path = pathSlice.slice(2)?.join("/") || "";
+
+  return (
+    <div style={{ display: "flex", alignItems: "center", gap: "10px" }}>
```
Comment (alfonsograziano) [inline apps/payments/app/components/Header/LanguageSwitch.tsx:9]: Is this code duplication? It looks to me that we have the language switch into two places (here and in `HamburgerMenu.tsx`)

**Review by alfonsograziano — APPROVED** [review]
Apart from a component that might need refactoring, LGTM

File: apps/payments/app/components/HamburgerMenu/HamburgerMenu.module.scss [hunk 3]
```diff
@@ -0,0 +1,90 @@
+@import "/app/styles/globals.scss";
```
[the remark this answers is not in the document]
Comment (peschina) [reply apps/payments/app/components/HamburgerMenu/HamburgerMenu.module.scss:1]: For sure. Our implementation is very similar to the one in web, but not quite the same. A bit simpler and leaner I hope :)

**Pull request merged**